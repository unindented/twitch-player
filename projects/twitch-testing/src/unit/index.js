import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
