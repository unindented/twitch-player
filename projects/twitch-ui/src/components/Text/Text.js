import PropTypes from "prop-types";
import React, { memo } from "react";
import { Text } from "react-native";
import { useTheme } from "../../hooks";

const CustomText = memo(({ style = [], ...props }) => {
  const { typography } = useTheme();
  const fontFamily = typography.family;
  const compositeStyle = [].concat(style).concat({ fontFamily });

  return <Text {...props} style={compositeStyle} />;
});

CustomText.propTypes = {
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.arrayOf(PropTypes.object),
  ]),
};

export default CustomText;
