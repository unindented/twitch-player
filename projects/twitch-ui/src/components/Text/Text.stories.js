import { storiesOf } from "@storybook/react";
import React from "react";
import Text from ".";

storiesOf("Components|Text", module).add("default", () => <Text>OHAI</Text>);
