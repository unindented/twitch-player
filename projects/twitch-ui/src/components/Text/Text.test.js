import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import Text from "./Text";

describe("Text", () => {
  let instance;

  beforeEach(() => {
    instance = render(<Text testID="text">OHAI</Text>);
  });

  it("renders the text", () => {
    expect(instance.getByTestId("text")).toMatchSnapshot();
  });
});
