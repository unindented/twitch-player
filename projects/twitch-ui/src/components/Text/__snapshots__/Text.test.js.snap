// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders the text 1`] = `
<div
  class="css-text-76zvg2"
  data-testid="text"
  dir="auto"
  style="font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Helvetica Neue',sans-serif; line-height: 1.5;"
>
  OHAI
</div>
`;
